node/pkg/watchers/evm/connectors/batch_poller.go

```diff
@@ -0,0 +1,285 @@
+type BatchPollConnector struct {
+	Connector
+	Delay     time.Duration
+	blockFeed ethEvent.Feed
+	errFeed   ethEvent.Feed
+	batchData []BatchEntry
+}
+
+type (
+	Blocks []*NewBlock
+
+	BatchEntry struct {
+		tag      string
+		finality FinalityLevel
+	}
+
+	BatchResult struct {
+		result BlockMarshaller
+		err    error
+	}
+)
+
+const MAX_GAP_BATCH_SIZE uint64 = 5
+
+func NewBatchPollConnector(ctx context.Context, baseConnector Connector, delay time.Duration) (*BatchPollConnector, error) {
+	// Create the batch data in the order we want to report them to the watcher, so finalized is most important, latest is least.
+	batchData := []BatchEntry{
+		{tag: "finalized", finality: Finalized},
+		{tag: "safe", finality: Safe},
+		{tag: "latest", finality: Latest},
+	}
+
+	connector := &BatchPollConnector{
+		Connector: baseConnector,
+		Delay:     delay,
+		batchData: batchData,
+	}
+	err := supervisor.Run(ctx, "batchPoller", common.WrapWithScissors(connector.runFromSupervisor, "batchPoller"))
+	if err != nil {
+		return nil, err
+	}
+	return connector, nil
+}
+
+func (b *BatchPollConnector) SubscribeForBlocks(ctx context.Context, errC chan error, sink chan<- *NewBlock) (ethereum.Subscription, error) {
+	sub := NewPollSubscription()
+	blockSub := b.blockFeed.Subscribe(sink)
+
+	// The feed library does not support error forwarding, so we're emulating that using a custom subscription and
+	// an error feed. The feed library can't handle interfaces which is why we post strings.
+	innerErrSink := make(chan string, 10)
+	innerErrSub := b.errFeed.Subscribe(innerErrSink)
+
+	common.RunWithScissors(ctx, errC, "block_poll_subscribe_for_blocks", func(ctx context.Context) error {
+		for {
+			select {
+			case <-ctx.Done():
+				blockSub.Unsubscribe()
+				innerErrSub.Unsubscribe()
+				return nil
+			case <-sub.quit:
+				blockSub.Unsubscribe()
+				innerErrSub.Unsubscribe()
+				sub.unsubDone <- struct{}{}
+				return nil
+			case v := <-innerErrSink:
+				sub.err <- fmt.Errorf(v)
+			}
+		}
+	})
+	return sub, nil
+}
+
+func (b *BatchPollConnector) runFromSupervisor(ctx context.Context) error {
+	logger := supervisor.Logger(ctx).With(zap.String("eth_network", b.Connector.NetworkName()))
+	supervisor.Signal(ctx, supervisor.SignalHealthy)
+	return b.run(ctx, logger)
+}
+
+func (b *BatchPollConnector) run(ctx context.Context, logger *zap.Logger) error {
+	// Get the initial blocks.
+	lastBlocks, err := b.getBlocks(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	timer := time.NewTimer(b.Delay)
+	defer timer.Stop()
+
+	errCount := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-timer.C:
+			lastBlocks, err = b.pollBlocks(ctx, logger, lastBlocks)
+			if err != nil {
+				errCount++
+				logger.Error("batch polling encountered an error", zap.Int("errCount", errCount), zap.Error(err))
+				if errCount > 3 {
+					b.errFeed.Send(fmt.Sprint("polling encountered an error: ", err))
```

Comment:
Do you want to clear the error count so you send on every 3rd or just every one after 3?